src/test/java/edu/ucsb/courses/controllers/ScheduleControllerTests.java

```diff
@@ -0,0 +1,334 @@
+package edu.ucsb.courses.controllers;
+
+
+import edu.ucsb.courses.advice.AuthControllerAdvice;
+import edu.ucsb.courses.entities.AppUser;
+import netscape.javascript.JSObject;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import edu.ucsb.courses.repositories.ScheduleRepository;
+import edu.ucsb.courses.entities.Schedule;
+
+@WebMvcTest(value = ScheduleController.class)
+@WithMockUser
+public class ScheduleControllerTests {
+
+  @Autowired
+  private MockMvc mockMvc;
+
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  @MockBean
+  ScheduleRepository mockScheduleRepository;
+
+  @MockBean
+  AuthControllerAdvice authController;
+
+  private String userToken() {
+    return "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTYiLCJuYW1lIjoiSm9obiBEb2UiLCJpYXQiOjE1MTYyMzkwMjJ9.MkiS50WhvOFwrwxQzd5Kp3VzkQUZhvex3kQv-CLeS3M";
+  }
+
+  @Test
+    public void testUpdateSchedule_scheduleExists_updateValues() throws Exception {
+      Schedule inputSchedule = new Schedule(1L,"CS 156", "New", "Fall 2020", "123456");
+      Schedule savedSchedule = new Schedule(1L,"CS 156", "Old", "Fall 2020", "123456");
+      String body = objectMapper.writeValueAsString(inputSchedule);
+      AppUser user = new AppUser();
+      user.setId(123456L);
+      when(authController.getUser(any(String.class))).thenReturn(user);
+
+      when(mockScheduleRepository.findById(any(Long.class))).thenReturn(Optional.of(savedSchedule));
+      when(mockScheduleRepository.save(inputSchedule)).thenReturn(inputSchedule);
+      MvcResult response =
+          mockMvc
+              .perform(put("/api/public/updateSchedule?id=1&name=CS 156&description=New&quarter=Fall 2020&userId=123456").with(csrf()).contentType(MediaType.APPLICATION_JSON)
```

Comment:
Change `/api/public/` to `/api/member/` throughout the file (to match) change in the controller.